kratos/geometries/line_2d_2.h

```diff
@@ -876,47 +876,38 @@ class Line2D2 : public Geometry<TPointType>
         VectorPoint[1] = rPoint[1] - FirstPoint[1];
         const double dist_proy = VectorPoint[0] * Normal[0] + VectorPoint[1] * Normal[1];
 
-//        KRATOS_WATCH(rPoint);
-//        KRATOS_WATCH(Point_projected);
-//        KRATOS_WATCH(dist_proy);
+        const CoordinatesArrayType ProjectedPoint = rPoint - Normal * dist_proy;
 
-        if (dist_proy < tol)
-        {
-            const double L  = Length();
-
-            const double l1 = std::sqrt((rPoint[0] - FirstPoint[0]) * (rPoint[0] - FirstPoint[0])
-                      + (rPoint[1] - FirstPoint[1]) * (rPoint[1] - FirstPoint[1]));
+        const double L  = Length();
 
-            const double l2 = std::sqrt((rPoint[0] - SecondPoint[0]) * (rPoint[0] - SecondPoint[0])
-                      + (rPoint[1] - SecondPoint[1]) * (rPoint[1] - SecondPoint[1]));
+        const double l1 = std::sqrt((ProjectedPoint[0] - FirstPoint[0]) * (ProjectedPoint[0] - FirstPoint[0])
+                    + (ProjectedPoint[1] - FirstPoint[1]) * (ProjectedPoint[1] - FirstPoint[1]));
 
-//            std::cout << "L: " << L << " l1: " << l1 << " l2: " << l2 << std::endl;
-
-            if (l1 <= (L + tol) && l2 <= (L + tol))
-            {
-                rResult[0] = 2.0 * l1/(L + tol) - 1.0;
-            }
-            else if (l1 > (L + tol))
-            {
-                rResult[0] = 2.0 * l1/(L + tol) - 1.0; // NOTE: The same value as before, but it will be > than 1
-            }
-            else if (l2 > (L + tol))
-            {
-                rResult[0] = 1.0 - 2.0 * l2/(L + tol);
-            }
-            else
-            {
-                rResult[0] = 2.0; // Out of the line!!! TODO: Check if this value gives problems
-            }
+        const double l2 = std::sqrt((ProjectedPoint[0] - SecondPoint[0]) * (ProjectedPoint[0] - SecondPoint[0])
+                    + (ProjectedPoint[1] - SecondPoint[1]) * (ProjectedPoint[1] - SecondPoint[1]));
 
+        if (l1 <= (L + tol) && l2 <= (L + tol))
+        {
+            rResult[0] = 2.0 * l1/(L + tol) - 1.0;
+        }
+        else if (l1 > (L + tol))
+        {
+            rResult[0] = 2.0 * l1/(L + tol) - 1.0; // NOTE: The same value as before, but it will be > than 1
```

Comment:
You are doing the same operation as in the previous `if`. I would add this as an _or_ clause to avoid code duplication.